boat/client/src/test/java/com/example/RegisterControllerIT.java

```diff
@@ -0,0 +1,65 @@
+package com.example;
+
+import com.backbase.buildingblocks.test.http.TestRestTemplateConfiguration;
+import com.backbase.greeting.api.service.v1.GreetingApi;
+import com.backbase.greeting.api.service.v1.model.GreetingPostResponse;
+import com.example.dto.RegisterUserRequestDTO;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(classes = BoatExampleClientApplication.class)
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+@AutoConfigureMockMvc
+@ActiveProfiles("it")
+class RegisterControllerIT {
```

Comment:
lets also mention this guy on the readme so a new joiner would have some reference on how to create Spring Boot Tests using the API